### PR TITLE
Always include a root node in the navigation bar.

### PR DESCRIPTION
**Pull request by andy-ms**
This should wait for #8764 to be resolved so that the root has all the children it needs.

### CODE REVIEW

**Comment by mhegazy** [comment]
:+1:

**Comment by andy-ms** [comment]
Added a new commit which needs review.

File: src/harness/fourslash.ts [hunk 1]
```diff
@@ -1959,71 +1959,47 @@ namespace FourSlash {
...
+            let items = this.languageService.getNavigationBarItems(this.activeFile.fileName);
+            items = this.simplifyNavigationBar(items);
+            if (JSON.stringify(items) !== JSON.stringify(json)) {
+                this.raiseError(`verifyNavigationBar failed - expected: ${JSON.stringify(json, undefined, 2)}, got: ${JSON.stringify(items, undefined, 2)}`);
+            }
+        }
+
+        // Remove any properties that tend to all have the same value so that test data is easier to read.
+        private simplifyNavigationBar(items: ts.NavigationBarItem[]): any {
+            return items.map(item => {
```
Comment (mhegazy) [inline src/harness/fourslash.ts]: if this is only for comparison purposes, i would do this in the replacer function for `JSON.stringify(items, (k,v)=> { if (key === "indent") return undefined; return v; });`

**Comment by mhegazy** [comment]
test changes look good. i would give the JSON.stringify(item, replacerFunction) a try.

**Pull request merged**